client/router_create.go

```diff
@@ -545,7 +543,6 @@ func (cli *VanClient) GetRouterSpecFromOpts(options types.SiteConfigSpec, siteId
 		Distribution: "multicast",
 	})
 	routerConfig.AddListener(qdr.Listener{
```

Comment:
Observing gettaddrinfo error in the logs, is this expected?

2022-03-23 12:20:36.322919 +0000 SERVER (info) HTTP server thread running
2022-03-23 12:20:36.323554 +0000 CONN_MGR (info) Configured Listener: localhost:5672 proto=any, role=normal
2022-03-23 12:20:36.323952 +0000 SERVER (error) getaddrinfo(, 9090) failed with Name or service not known
2022-03-23 12:20:36.323987 +0000 SERVER (notice) Disabling ipv6 on :9090
2022-03-23 12:20:36.324196 +0000 SERVER (notice) Listening for HTTP on :9090

Reply:
Define 'expected' :-)

This PR was based on reading the documentation for the `host` field in a `listener` (https://qpid.apache.org/releases/qpid-dispatch-1.19.0/man/qdrouterd.conf.html#_listener):

```
A host name, IPV4 or IPV6 literal, or the empty string. The empty string listens on all local addresses. A host name listens on all addresses associated with the name. An IPV6 literal address (or wildcard [::]) listens only for IPV6. An IPV4 literal address (or wildcard 0.0.0.0) listens only for IPV4. 
```

However listeners where http=true use libwebsockets rather than proton, and that is where this error originates. There is code there, recently introduced, that tries to determine if ipv6 is enabled on the interface specified or not. That check does not handle the empty string. As well as the error above, this means that the http listener (through which the router console is served and management operations are sent) will disable ipv6.

Either that check needs to be altered, or the http listeners will need to specify an explicit host. Will dig a bit more to decide which path makes most sense,

Reply:
sorry, 'expected' was a nonsensical work to use ;-)

Reply:
word

Reply:
I think the right fix is in the router: https://github.com/skupperproject/skupper-router/pull/218